src/pages/VeSDL/VeTokenCalculator.tsx

```diff
@@ -35,33 +34,60 @@ export default function VeTokenCalculator({
   const { t } = useTranslation()
   const basicPools = useContext(BasicPoolsContext)
   const gaugeData = useContext(GaugeContext)
+  const [userLPAmountInput, setUserLPAmountInput] = useState<string>("")
+  const [totalLPAmountInput, setTotalLPAmountInput] = useState<string>("")
   const [poolNameValue, setPoolNameValue] = useState<string>("D4")
-  const [userVeSdlInputAmount, setUserVeSdlInputAmount] =
-    useState<BigNumber>(Zero)
+  const [userVeSdlInputAmount, setUserVeSdlInputAmount] = useState<string>(
+    userBalanceVeSdl.isZero() ? "" : formatBNToString(userBalanceVeSdl, 18),
+  )
+  const [totalVeSDLInput, setTotalVeSDLInput] = useState<string>(
+    totalSupplyVeSdl.isZero() ? "" : formatBNToString(totalSupplyVeSdl, 18),
+  )
+
   const pool = basicPools && basicPools[poolNameValue]
   const gauge =
-    !!pool?.poolAddress &&
-    gaugeData.gauges &&
-    gaugeData.gauges[pool.poolAddress]
+    !!pool?.poolAddress && gaugeData.gauges
+      ? gaugeData.gauges[pool.poolAddress]
+      : undefined
 
-  const userVeSdlAmount = userVeSdlInputAmount.isZero()
-    ? userBalanceVeSdl
-    : userVeSdlInputAmount
+  const userLPAmountBN = parseEther(userLPAmountInput || "0")
+  const totalLPAmountBN = parseEther(totalLPAmountInput || "0")
 
-  const userLPAmount = gauge ? gauge.gaugeBalance : Zero
-  const totalLPAmount = gauge ? gauge.gaugeTotalSupply : Zero
-  const totalWorkingSupply = gauge ? gauge.workingSupply : Zero
+  const boost =
+    gauge &&
+    calculateWorkingAmountAndBoost(
+      userLPAmountBN,
+      totalLPAmountBN,
+      gauge.workingBalances,
+      gauge.workingSupply,
+      parseEther(userVeSdlInputAmount || "0"),
+      parseEther(totalVeSDLInput || "0"),
+    )
 
-  const { boost } = calculateWorkingAmountAndBoost(
-    userLPAmount,
-    totalLPAmount,
-    totalWorkingSupply,
-    userVeSdlAmount,
-    totalSupplyVeSdl,
-  )
-  const veSdlForMaxBoost =
-    totalLPAmount.gt(Zero) &&
-    userLPAmount.mul(totalSupplyVeSdl).div(totalLPAmount)
+  const minVeSDL =
+    !totalLPAmountBN.add(userLPAmountBN).isZero() &&
+    parseEther(totalVeSDLInput || "0")
+      .mul(userLPAmountBN)
+      .div(totalLPAmountBN.add(userLPAmountBN))
+
+  const maxBoostPossible =
+    gauge &&
+    minVeSDL &&
+    calculateWorkingAmountAndBoost(
+      userLPAmountBN,
+      totalLPAmountBN,
+      gauge.workingBalances,
+      gauge.workingSupply,
+      minVeSDL,
+      parseEther(totalVeSDLInput || "0"),
```

Comment:
Do you worry the calculation will be off if there's a chance value can be Zero? Is it possible to null check `totalVeSDLInput` higher up or is this approach safe?

Reply:
I think parseEther don't gives error when `totalVeSDLInput `is empty string.

Reply:
As long as there's no scenario where it's going to default to "0" and give the wrong result for maxBoostPossible

Reply:
When `totalVeSDLInput` is null, the boost value should have 1 and  when totalVeSDLInput is Zero, it returns also 1.
So,  I think we can give Zero when it is null.

But as Duy suggested, I'll check the votingEscrowContract and gauge to avoid the `null` value.

Reply:
Ok thanks for explaining.